src/users/users.service.ts

```diff
@@ -1,35 +1,69 @@
 import { Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
-import { SetSshDto } from './dto/set-ssh.dto';
+import { SetSshDto } from 'src/users/dto/set-ssh.dto';
 import { PrismaService } from 'src/prisma.service';
 import { User, Prisma } from '@prisma/client';
 import { SanitizedUser } from './types/sanitized-user.type';
 import { exclude } from 'src/utils/prisma-exclude';
+import * as bcrypt from 'bcrypt';
+import { EMAIL_REGEX, PASSWORD_REGEX, USERNAME_REGEX } from 'src/utils/constants';
 
 @Injectable()
 export class UsersService {
   constructor(private prisma: PrismaService) {}
 
-  private passwd_encrypt(password: string) {
-    // FIXME: implement encrypting passwords
-    return password;
-  }
-
   private sanitizeOutput(user: User): SanitizedUser {
     return exclude(user, ['id', 'email_nonce', 'password', 'createdAt', 'updatedAt']);
   }
 
+  private async passwd_encrypt(password: string): Promise<string> {
+    const saltOrRounds = await bcrypt.genSalt(10);
+    const hash = await bcrypt.hash(password, saltOrRounds);
+    return hash;
+  }
+
   async create(user: CreateUserDto): Promise<SanitizedUser> {
     return this.sanitizeOutput(await this.prisma.user.create({
       data: {
         username: user.username,
         email: user.email,
-        password: this.passwd_encrypt(user.password),
-        isAdmin: false,
+        password: await this.passwd_encrypt(user.password),
+        isAdmin: true,
```

Comment:
Users should definitely not be admins by default. I think this is something that's been forgotten during a commit.

Reply:
Yep, well noticed, I forgot to change it back, thanks.